src/strategies/jpegd-locked-jpeg-of/index.ts

```diff
@@ -0,0 +1,71 @@
+import { formatUnits } from '@ethersproject/units';
+import { getAddress } from '@ethersproject/address';
+import { multicall } from '../../utils';
+import { subgraphRequest } from '../../utils';
+
+export const author = '0xleez';
+export const version = '0.1.0';
+
+const UNISWAP_SUBGRAPH_URL = {
+  '1': 'https://api.thegraph.com/subgraphs/name/jpegd/jpegd-core-mainnet'
+};
+
+const abi = [
+  'function traitBoostPositions(uint256 _nftIndex) view returns (address owner, uint256 unlockAt, uint256 lockedValue)',
+  'function ltvBoostPositions(uint256 _nftIndex) view returns (address owner, uint256 unlockAt, uint256 lockedValue)'
+];
+
+export async function strategy(
+  _space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const { collectionToProviderAddress } = options;
+
+  const params = {
+    jpeglocks: {
+      __args: {
+        where: {
+          owner_in: addresses.map((address) => address.toLowerCase())
+        }
+      },
+      owner: { id: true },
+      collection: { id: true },
+      type: true,
+      nftIndex: true,
+      amount: true,
+      unlockTime: true
+    }
+  };
+
+  const result = await subgraphRequest(UNISWAP_SUBGRAPH_URL[network], params);
```

Comment:
Make sure to pass `snapshot` to `_args.block` else the results will be unpredictable